prover/examples/zkmips.rs

```diff
@@ -340,6 +427,7 @@ fn main() {
         "prove_groth16" => prove_groth16(),
         "bench" => prove_sha2_bench(),
         "revm" => prove_revm(),
+        "goexample" => prove_add_example(),
```

Comment:
We should refector this parameter. We should not mix the step with the specific host program. we can use a new parameter to represent the host program.